Mage.Sets/src/mage/cards/e/EverythingComesToDust.java

```diff
@@ -0,0 +1,70 @@
+package mage.cards.e;
+
+import mage.MageObjectReference;
+import mage.abilities.effects.common.ExileAllEffect;
+import mage.abilities.keyword.ConvokeAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.FilterPermanent;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.util.CardUtil;
+
+import java.util.HashSet;
+import java.util.UUID;
+
+/**
+ *
+ * @author notgreat
+ */
+public final class EverythingComesToDust extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent("creatures except those that share a creature type with a creature that convoked this spell, all artifacts, and all enchantments");
+
+    static {
+        filter.add(EverythingComesToDustPredicate.instance);
+    }
+    public EverythingComesToDust(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{7}{W}{W}{W}");
+
+        // Convoke
+        this.addAbility(new ConvokeAbility());
+
+        // Exile all creatures except those that share a creature type with a creature that convoked this spell, all artifacts, and all enchantments.
+        this.getSpellAbility().addEffect(new ExileAllEffect(filter));
+    }
+
+    private EverythingComesToDust(final EverythingComesToDust card) {
+        super(card);
+    }
+
+    @Override
+    public EverythingComesToDust copy() {
+        return new EverythingComesToDust(this);
+    }
+}
+enum EverythingComesToDustPredicate implements ObjectSourcePlayerPredicate<Permanent> {
+    instance;
+    @Override
+    public boolean apply(ObjectSourcePlayer<Permanent> input, Game game) {
+        Permanent p = input.getObject();
+        if (CardType.ARTIFACT.getPredicate().apply(p, game) || CardType.ENCHANTMENT.getPredicate().apply(p, game)){
+            return true;
+        }
+        if (!CardType.CREATURE.getPredicate().apply(p, game)){
```

Comment:
is there a reason for this verbose style rather than the simpler `p.isCreature(game)` ?

Reply:
Remnant from my attempts of making a `Predicates.Or` version of the filter; the simpler one is definitely better here.